[COMPRESS-714] Internal IllegalArgumentException in ZipFile and TarFile creation is not caught

garydgregory opened the pull request:
Fixes Jira https://issues.apache.org/jira/browse/COMPRESS-714

Before you push a pull request, review this list:

- [x] Read the [contribution guidelines](CONTRIBUTING.md) for this project.
- [x] Read the [ASF Generative Tooling Guidance](https://www.apache.org/legal/generative-tooling.html) if you use Artificial Intelligence (AI).
- [ ] I used AI to create any part of, or all of, this pull request.
- [x] Run a successful build using the default [Maven](https://maven.apache.org/) goal with `mvn`; that's `mvn` on the command line by itself.
- [x] Write unit tests that match behavioral changes, where the tests fail if the changes to the runtime are not applied. This may not always be possible, but it is a best practice.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Each commit in the pull request should have a meaningful subject line and body. Note that a maintainer may squash commits during the merge process.

garydgregory committed:
[COMPRESS-714] Internal IllegalArgumentException in ZipFile and TarFile
creation is not caught

garydgregory committed:
The bug fix for [COMPRESS-714] is better done in Commons IO

garydgregory committed:
TEMP in PR until IO is released.

garydgregory committed:
The bug fix for [COMPRESS-714] is better done in Commons IO

garydgregory committed:
The bug fix for [COMPRESS-714] is better done in Commons IO

garydgregory committed:
Use IOUtils.close[Quietly]()

ppkarwasz commented on src/main/java/org/apache/commons/compress/archivers/sevenz/SevenZFile.java:
```diff
@@ -629,9 +629,7 @@ private SevenZFile(final Builder builder) throws IOException {
-            IOUtils.close(channel, archiveException::addSuppressed);
-            throw archiveException;
+            throw IOUtils.closeQuietly(channel, new ArchiveException(e));
         }
```
Honestly, I wouldn't delay the release of `commons-compress` version `1.29.0` more than it already has been to wait for a new version of `commons-io`.

Let's use what we have in `2.21.0`:

```java
        } catch (final Exception e) {
            final IOException ioe = e instanceof IOException ? (IOException) e : new ArchiveException(e);
            IOUtils.closeQuietly(channel, ioe::addSuppressed);
            throw ioe;
        }
```

When we **do** release Commons IO version `2.22.0`, we'll update this call site and **all** the other call sites that might benefit from the new method. There is no chance to forget that, since `closeQuietly` becomes ambiguous in IO `2.22.0`.

jtattermusch commented:
> @@aisle-analyzer experiment.

it's supposed to be @aisle-analyzer, can you please try again?

ideally in a new comment.

garydgregory committed:
Pick up Commons IO 2.22.0

garydgregory committed:
Merge branch 'master' into bugifx/COMPRESS-714

garydgregory committed:
Removing temporary Maven repository and fix tag

Removed temporary repository configuration and cleaned up dependency closing tag.
